### DISCUSSION TITLE
Is it possible to embed a custom font and apply that custom font to the app?

### DISCUSSION THREAD

**Discussion by santokalayil** [fltk-rs/fltk-rs#713] [General]
Is it possible to embed a custom font and apply that custom font to the app?  I mean not the system fonts, but fonts downloaded from google fonts. Also is there a way to embed an icon from material design fonts so that they can be used button icons in apps?

**Comment by MoAlyousef** [comment]
You can use fltk::app::App::load_font(). The docs have an example on how to use it.

**Reply by santokalayil** [reply]
thank you very much

**Reply by santokalayil** [reply]
I think loading custom external font is not possible. The load_font() can be used to select font from one of the system fonts installed default in the system. 
I tried the code in the docs:
```
    let font = app.load_font("assets\\fonts\\StintUltraCondensed-Regular.ttf").unwrap();
    let mut btn = fltk::button::Button::new(100, y_pos, 500, 25, "Hi Good Morining");
    btn.set_label_font(fltk::enums::Font::by_index(9));
    btn.set_label_size(30);
    btn.set_color(fltk::enums::Color::Red);
    btn.set_label_font(fltk::enums::Font::by_name(&font));
```
It runs without error but the font is the default font, and not the `StintUltraCondensed-Regular.ttf` font

**Reply by MoAlyousef** [reply]
FLTK requires the font name to be able to load it, and even then it only allows a slot for an extra loaded font. Some google fonts eschew that info and using stb's stbtt_GetFontNameString just returns null. If you need to bundle a font, you'll have to check it works beforehand, otherwise, you can use a drawing lib to rasterize the font into an image and use the image on a widget, and you can use as many fonts as you wish.
